### PR TITLE
CSHARP-763 Execution Profiles with Mapper and LINQ APIs

### PR DESCRIPTION
**Pull request by joao-r-reis**

### CODE REVIEW

File: src/Cassandra/Data/Linq/CqlQuerySingleElement.cs [hunk 1]
```diff
@@ -77,9 +83,19 @@ public new TEntity EndExecute(IAsyncResult ar)
         /// </summary>
         public new TEntity Execute()
         {
-            var queryAbortTimeout = GetTable().GetSession().GetConfiguration()?.ClientOptions.QueryAbortTimeout ?? ClientOptions.DefaultQueryAbortTimeout;
+            var queryAbortTimeout = GetTable().GetSession().Cluster.Configuration.DefaultRequestOptions.QueryAbortTimeout;
```
Comment (joao-r-reis) [inline src/Cassandra/Data/Linq/CqlQuerySingleElement.cs]: https://github.com/datastax/csharp-driver/pull/442#discussion_r278009124

**Comment by joao-r-reis** [comment]
Rebased after #442 merge.

File: src/Cassandra/Data/Linq/Batch.cs [hunk 2]
```diff
@@ -68,15 +68,29 @@ public Batch Append(IEnumerable<CqlCommand> cqlCommands)
 
         public void Execute()
         {
-            EndExecute(BeginExecute(null, null));
+            var queryAbortTimeout = _session.Cluster.Configuration.DefaultRequestOptions.QueryAbortTimeout;
```
Comment (jorgebay) [inline src/Cassandra/Data/Linq/Batch.cs]: We could use this opportunity to make an internal reusable property whenever possible.

File: src/Cassandra/Data/Linq/CqlScalar.cs [hunk 3]
```diff
@@ -43,6 +43,13 @@ public new TEntity Execute()
             var task = ExecuteAsync();
             return TaskHelper.WaitToComplete(task, queryAbortTimeout);
         }
+        
+        public new TEntity Execute(string executionProfile)
+        {
+            var queryAbortTimeout = GetTable().GetSession().Cluster.Configuration.DefaultRequestOptions.QueryAbortTimeout;
```
Comment (jorgebay) [inline src/Cassandra/Data/Linq/CqlScalar.cs]: Same.

File: src/Cassandra/Data/Linq/CqlQuerySingleElement.cs [hunk 4]
```diff
@@ -81,5 +87,15 @@ public new TEntity Execute()
             var task = ExecuteAsync();
             return TaskHelper.WaitToComplete(task, queryAbortTimeout);
         }
+        
+        /// <summary>
+        /// Evaluates the Linq query, executes the cql statement with the provided execution profile and returns the first result.
+        /// </summary>
+        public new TEntity Execute(string executionProfile)
+        {
+            var queryAbortTimeout = GetTable().GetSession().Cluster.Configuration.DefaultRequestOptions.QueryAbortTimeout;
```
Comment (jorgebay) [inline src/Cassandra/Data/Linq/CqlQuerySingleElement.cs]: Same.

File: src/Cassandra/Data/Linq/CqlQueryBase.cs [hunk 5]
```diff
@@ -143,5 +165,21 @@ public IEnumerable<TEntity> EndExecute(IAsyncResult ar)
             var task = (Task<IEnumerable<TEntity>>)ar;
             return task.Result;
         }
+        
+        private IEnumerable<TEntity> ExecuteCqlQuery(string executionProfile)
+        {
+            var queryAbortTimeout = GetTable().GetSession().Cluster.Configuration.DefaultRequestOptions.QueryAbortTimeout;
```
Comment (jorgebay) [inline src/Cassandra/Data/Linq/CqlQueryBase.cs]: Same :)

File: src/Cassandra/Data/Linq/Batch.cs (hunk 2 above)
Comment (joao-r-reis) [reply src/Cassandra/Data/Linq/Batch.cs]: I added a property here but it's not reusable anywhere else currently.

File: src/Cassandra/Data/Linq/CqlScalar.cs (hunk 3 above)
Comment (joao-r-reis) [reply src/Cassandra/Data/Linq/CqlScalar.cs]: done

File: src/Cassandra/Data/Linq/CqlQuerySingleElement.cs (hunk 4 above)
Comment (joao-r-reis) [reply src/Cassandra/Data/Linq/CqlQuerySingleElement.cs]: done

File: src/Cassandra/Data/Linq/CqlQueryBase.cs (hunk 5 above)
Comment (joao-r-reis) [reply src/Cassandra/Data/Linq/CqlQueryBase.cs]: done

File: src/Cassandra/Data/Linq/CqlQuerySingleElement.cs (hunk 1 above)
Comment (joao-r-reis) [reply src/Cassandra/Data/Linq/CqlQuerySingleElement.cs]: done

**Pull request merged**